tests/unit/test_blob.py

```diff
@@ -1024,6 +1021,99 @@ def _mock_requests_response(status_code, headers, content=b""):
         response.request = requests.Request("POST", "http://example.com").prepare()
         return response
 
+    def test__extract_headers_from_download_gzipped(self):
```

Comment:
Adding coverage, too?

Reply:
The tests were misnamed, and out of place:  I just fixed that, and split a couple of over-large ones.